demo/app/ios/Runner/flutterPlugin.swift

```diff
@@ -403,20 +411,103 @@ public class SwiftWalletSDKPlugin: NSObject, FlutterPlugin {
                                              details: "Pass issuerURI as the arguments"))
         }
                 
-    
+        guard let walletSDK = self.walletSDK else{
+            return  result(FlutterError.init(code: "NATIVE_ERR",
+                                             message: "error while creating new OIDC interaction",
+                                             details: "WalletSDK interaction is not initialized, call initSDK()"))
+        }
+        
         do {
-            let walletInitiatedOpenID4CI = try walletSDK?.createOpenID4CIWalletIntiatedInteraction(issuerURI: issuerURI)
-            //TODO- VCS part is not implemented yet, this logic needs to be updated to get the create authorization URL
-            try walletInitiatedOpenID4CI?.getSupportedCredentials()
 
+            let walletInitiatedOpenID4CI = try walletSDK.createOpenID4CIWalletIntiatedInteraction(issuerURI: issuerURI)
+            let supportedCredentials = try walletInitiatedOpenID4CI.getSupportedCredentials()
+            
+            var supportedCredentialsList: [Any] = []
+            for index in 0..<supportedCredentials.length() {
+                var typeStrArray = [String]()
+                for i in 0..<(supportedCredentials.atIndex(index)!.types()?.length())!{
+                    let type1 = supportedCredentials.atIndex(index)!.types()?.atIndex(i)
+                    typeStrArray.append(type1!)
+                }
+                
+                let supportedCredentialResp:[String:Any] = [
+                    "format":  supportedCredentials.atIndex(index)!.format(),
+                    "types":   typeStrArray
+                ]
+                supportedCredentialsList.append(supportedCredentialResp)
+            }
+            self.walletInitiatedOpenID4CI = walletInitiatedOpenID4CI
+            result(supportedCredentialsList)
+        
         } catch {
             result(FlutterError.init(code: "NATIVE_ERR",
                                      message: "error while initializing wallet initiated issuance flow",
                                      details: error))
         }
         
     }
-    
+     
+    public func createAuthorizationUrlWalletInitiatedFlow(arguments: Dictionary<String, Any>, result: @escaping FlutterResult){
+        guard let walletInitiatedOpenID4CI = self.walletInitiatedOpenID4CI else{
+            return  result(FlutterError.init(code: "NATIVE_ERR",
+                                             message: "error while process wallet initiated flow",
```

Comment:
Not sure exactly what `error while process wallet initiated flow` means... I would replace it with your `details` message message below and then just leave `details` blank.